### PR TITLE
Add more .gitignore entries

### PR DESCRIPTION
**Pull request by cmarqu**

### CODE REVIEW

**Comment by themperek** [comment]
Can you also add `.tox` folder and there are some files in the `endian_swapper` example generated (try run `tox`).

File: .gitignore [hunk 1]
```diff
@@ -64,3 +67,8 @@ waves.shm/
 
 # Interface libraries built on setup
 cocotb/libs
+pip-wheel-metadata/
+
+# Mentor Modelsim/Questa
```
Comment (themperek) [inline .gitignore]: `modelsim.ini and transcript are also in examples maybe block globally

Comment (themperek) [reply .gitignore]: also `*.wlf`

Comment (themperek) [reply .gitignore]: also some `*.json` files

Comment (cmarqu) [reply .gitignore]: > `modelsim.ini` and `transcript` are also in examples maybe block globally

I think a user can add local ``modelsim.ini`` files to define libraries etc., so ignoring them globally may be bad.

Comment (cmarqu) [reply .gitignore]: > also some `*.json` files

Globally, this may be too much. Is this specific to wavedrom maybe?

**Comment by cmarqu** [comment]
> Can you also add `.tox` folder and there are some files in the `endian_swapper` example generated (try run `tox`).

Added .tox, and the ``endian_swapper`` ignores in its own file in that directory.

File: .gitignore [hunk 2]
```diff
@@ -64,3 +68,11 @@ waves.shm/
...
+tests/test_cases/*/modelsim.ini
+tests/test_cases/*/transcript
+examples/test_cases/*/modelsim.ini
```
Comment (themperek) [inline .gitignore]: `examples/*/tests/modelsim.ini` ?

Comment (cmarqu) [reply .gitignore]: Good catch, fixed.

File: examples/endian_swapper/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+cosim/endian_swapper_hal_wrap.c
+cosim/hal.py
```
Comment (themperek) [inline examples/endian_swapper/.gitignore]:
```suggestion
cosim/hal.py
tests/wavedrom.json
```

Comment (cmarqu) [reply examples/endian_swapper/.gitignore]: Added.

**Pull request merged**